js/components/contact_download.jsx

```diff
@@ -0,0 +1,165 @@
+    });
+    requestActions.fetchAgencyFinderData(includeReferenceFields);
+  }
+
+  /**
+   * Download the CSV file.
+   */
+  performDownload() {
+    return ExcellentExport.convert({
+      anchor: this.downloadButton,
+      filename: 'FOIA-Agency-Contacts',
+      format: 'csv',
+    }, [{
+      name: 'FOIA Agency Contacts',
+      from: {
+        array: this.rows,
+      },
+    }]);
+  }
+
+  /**
+   * Event handler for when the download button is triggered.
+   */
+  handleClick() {
+    return (this.rows) ? this.performDownload() : false;
+  }
+
+  /**
+   * Called each time a batch of agency components finished loading.
+   */
+  render() {
+    const progress = agencyComponentStore.getState().agencyFinderDataProgress;
+
+    // If this is not the last batch, just update the button text.
+    if (progress < 100) {
+      if (this.downloadButton) {
+        this.downloadButton.innerHTML = `Please wait - ${progress}%`;
+      }
+    } else {
+      // Otherwise, prepare the CSV data.
+      const { agencyComponents } = this.state;
+
+      // Construct a large array of all the FOIA personnel as rows (objects).
+      const header = [
+        'Component',
+        'Department',
+        'Name',
+        'Title',
+        'Email',
+        'Street Address 1',
+        'Street Address 2',
+        'City',
+        'State',
+        'Zip code',
+        'Telephone',
+        'Fax',
+      ];
+      const rows = [];
+      agencyComponents.forEach((agencyComponent) => {
+        ContactDownloadButton.getFoiaPersonnelFields().forEach((personnelField) => {
+          agencyComponent[personnelField].forEach((foiaPersonnel) => {
+            rows.push([
+              agencyComponent.title,
+              agencyComponent.agency.name,
+              foiaPersonnel.name,
+              foiaPersonnel.title,
+              foiaPersonnel.email,
+              agencyComponent.submission_address.address_line1,
+              agencyComponent.submission_address.address_line2,
+              agencyComponent.submission_address.locality,
+              agencyComponent.submission_address.administrative_area,
+              agencyComponent.submission_address.postal_code,
+              (foiaPersonnel.phone) ? foiaPersonnel.phone.join(', ') : '',
+              agencyComponent.submission_fax,
+            ]);
+          });
+        });
+      });
+
+      // Sort the array of rows by the agency name, then component, then person name.
+      const orderPositions = [
+        header.indexOf('Department'),
+        header.indexOf('Component'),
+        header.indexOf('Name'),
+      ];
+      rows.sort((a, b) => {
+        let i;
+        for (i = 0; i < orderPositions.length; i++) {
+          const pos = orderPositions[i];
+          const aLower = (a[pos]) ? a[pos].toLowerCase() : '';
+          const bLower = (b[pos]) ? b[pos].toLowerCase() : '';
+          if (aLower < bLower) {
```

Comment:
You'll want to use `localeCompare` to make sure you're sorting in the user's locale.